consumer.go

```diff
@@ -96,11 +102,20 @@ func (c *Consumer) Run(ctx context.Context) {
 	if err != nil {
 		c.log.Printf("Error closing: %s", err)
 	}
+	c.terminated = true
+	c.mu.Unlock()
 	c.log.Println("Bye")
 }
 
 func (c *Consumer) Poll(timeoutMS int) (*rdkafka.Message, error) {
+	c.mu.Lock()
```

Comment:
We should probably  `defer c.mu.Unlock()` just after acquiring the lock so we only have one clear unlock path.
Releasing the lock after the `switch` statement does not have a performance hit, since we only do a log.